File: app/src/main/java/com/hitesh/todopod/itemsDatabase.java
package com.hitesh.todopod;

import android.content.Context;
import android.os.AsyncTask;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.sqlite.db.SupportSQLiteDatabase;

@Database(entities = {items.class}, version = 1)
public abstract class itemsDatabase extends RoomDatabase {
    private static itemsDatabase instance;
    public abstract itemsDAO itemsDAO();
    public static synchronized itemsDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    itemsDatabase.class, "note_database")
                    .fallbackToDestructiveMigration()
                    .addCallback(roomCallback)
                    .build();
        }
        return instance;
    }
    private static RoomDatabase.Callback roomCallback = new RoomDatabase.Callback() {
        @Override
        public void onCreate(@NonNull SupportSQLiteDatabase db) {
            super.onCreate(db);
            new PopulateDbAsyncTask(instance).execute();
        }
    };
    private static class PopulateDbAsyncTask extends AsyncTask<Void, Void, Void> {
        private itemsDAO itemsDao;
        private PopulateDbAsyncTask(itemsDatabase db) {
            itemsDao = db.itemsDAO();
        }
        @Override
        protected Void doInBackground(Void... voids) {
            return null;
        }
    }
}

File: app/src/main/java/com/hitesh/todopod/helper/itemsDatabase.kt
package com.hitesh.todopod.helper

import android.content.Context
import android.os.AsyncTask
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.sqlite.db.SupportSQLiteDatabase
import com.hitesh.todopod.daos.ItemsDAO
import com.hitesh.todopod.items

@Database(entities = [items::class], version = 1)
abstract class itemsDatabase : RoomDatabase() {
    abstract fun itemsDAO(): ItemsDAO
    private class PopulateDbAsyncTask(db: itemsDatabase?) : AsyncTask<Void?, Void?, Void?>() {

        init {
            val itemsDao: ItemsDAO = db!!.itemsDAO()
        }

        override fun doInBackground(vararg p0: Void?): Void? {
            return null
        }
    }

    companion object {
        private var instance: itemsDatabase? = null
        @Synchronized
        fun getInstance(context: Context): itemsDatabase? {
            if (instance == null) {
                instance = Room.databaseBuilder(context.applicationContext,
                        itemsDatabase::class.java, "note_database")
                        .fallbackToDestructiveMigration()
                        .addCallback(roomCallback)
                        .build()
            }
            return instance
        }

        private val roomCallback: Callback = object : Callback() {
            override fun onCreate(db: SupportSQLiteDatabase) {
                super.onCreate(db)
                PopulateDbAsyncTask(instance).execute()
            }
        }
    }
}